src/dda/dda.c

```diff
@@ -0,0 +1,58 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   dda.c                                              :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ycarro <ycarro@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/22 14:48:17 by ycarro            #+#    #+#             */
+/*   Updated: 2023/02/23 14:16:47 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "dda.h"
+
+void	dda(char **map, t_player *p, t_ray	*ray_data, t_point *collision)
+{
+	(void)map;
+	(void)collision;
+	set_incrementor(ray_data, ray_data->to_cast.radians);
+	get_ray_data(ray_data, ray_data->to_cast.radians, p);
+}
+
+void	set_incrementor(t_ray *ray_data, double alpha)
+{
+	ray_data->incrementor = ft_calloc(2, sizeof(double));
```

Comment:
i like to put all mem allocations in a ray_create() function and free them later in a ray_destroy() function
this way it is very easy to keep up with leaks
you can have a look at t_player for refrence

Reply:
I will try to change it, its true allocations should be controlled better

Reply:
Need to talk about this in the next meeting

Reply:
Changed